libs/sdk-ui-dashboard/src/model/store/filterContext/filterContextSelectors.ts

```diff
@@ -126,10 +126,20 @@ export const selectFilterContextDateFilter = createSelector(
  */
 export const selectFilterContextAttributeFilterByDisplayForm = memoize(
     (displayForm: ObjRef) =>
-        createSelector(selectFilterContextAttributeFilters, (attributeFilters) =>
-            attributeFilters.find((filter) =>
-                areObjRefsEqual(filter.attributeFilter.displayForm, displayForm),
-            ),
+        createSelector(
+            selectAttributeFilterDisplayFormsMap,
+            selectFilterContextAttributeFilters,
+            (attributeDisplayFormsMap, attributeFilters) => {
+                const df = attributeDisplayFormsMap.get(displayForm);
+                const dfUriRef = df && uriRef(df.uri);
+                const dfIdRef = df && idRef(df.id);
+
+                return attributeFilters.find(
+                    (filter) =>
+                        areObjRefsEqual(filter.attributeFilter.displayForm, dfUriRef) ||
+                        areObjRefsEqual(filter.attributeFilter.displayForm, dfIdRef),
+                );
```

Comment:
i suggest to go differently

1: modify/fix addAttributeFilterHandler, ensure that the native displayForm ref is used; the handler already validates new filter's display form, using the native ref is one-liner
2: get display form from map, then use the native ref again